lib/jara/releaser.rb

```diff
@@ -28,30 +28,34 @@ def initialize(environment, bucket_name, options={})
 
     def build_artifact
       if @environment.nil?
-        jar_name = "#{app_name}.jar"
+        archive_name = "#{app_name}.#{@archiver.extension}"
         Dir.chdir(project_dir) do
-          @archiver.create(jar_name: jar_name)
+          @archiver.create(archive_name: archive_name)
         end
         @logger.info('Created test artifact')
-        File.join(project_dir, 'build', jar_name)
+        File.join(project_dir, 'build', archive_name)
       elsif (artifact_path = find_local_artifact)
         @logger.warn('An artifact for %s already exists: %s' % [branch_sha[0, 8], File.basename(artifact_path)])
         artifact_path
       else
         date_stamp = Time.now.utc.strftime('%Y%m%d%H%M%S')
         destination_dir = File.join(project_dir, 'build', @environment)
-        jar_name = [app_name, @environment, date_stamp, branch_sha[0, 8]].join('-') << '.jar'
+        archive_name = [app_name, @environment, date_stamp, branch_sha[0, 8]].join('-') << '.' << @archiver.extension
         Dir.mktmpdir do |path|
           @shell.exec('git archive --format=tar --prefix=%s/ %s | (cd %s/ && tar xf -)' % [File.basename(path), branch_sha, File.dirname(path)])
           Dir.chdir(path) do
             @logger.info('Checked out %s from branch %s' % [branch_sha[0, 8], @branch])
-            @archiver.create(jar_name: jar_name)
+            if @build_command
+              @logger.info('Running build command: %s' % @build_command)
+              @shell.exec(@build_command)
+            end
```

Comment:
Yes, I see puck as a build tool, more than anything else. If we want to support systems not built using Puck, I think it is more reasonable to support any build-tool that creates an archive (where tar would be an extremely simple builder), instead of having to specify a combination (a command delegating to the build system) and an archiver in jara. Another possible example would be if the existing build system already produces a tarball/zipfile. It would be nice to be able to use that directly, rather than unpacking the files so that jara could repackage them.